src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -46,13 +47,15 @@ public void init() {
    * an empty array if the user is not provided.
    */
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException
+  {
 
     response.setContentType("application/json");
 
     String user = request.getParameter("user");
 
-    if (user == null || user.equals("")) {
+    if (user == null || user.equals(""))
```

Comment:
Use user.isEmpty() instead of user.equals(""). When reading the code, it is easier to read "if user is null or user is empty" rather than "if user is null or user is equal to no character".

Also, you probably should trim the whitespace off of user first. There is an edgecase where you might get a user like " " where there is a space in the middle and we don't want to count that as a user.

The code would look something like:

```
String user = request.getParameter("user").trim();

if (user == null || user.isEmpty()) {
    // ...
}
```